File: mobile-wallet-master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/transactions/ui/SpecificTransactionsActivity.java
package org.mifos.mobilewallet.mifospay.transactions.ui;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import org.mifos.mobilewallet.core.domain.model.Transaction;
import org.mifos.mobilewallet.mifospay.R;
import org.mifos.mobilewallet.mifospay.base.BaseActivity;
import org.mifos.mobilewallet.mifospay.receipt.ui.ReceiptActivity;
import org.mifos.mobilewallet.mifospay.transactions.TransactionsContract;
import org.mifos.mobilewallet.mifospay.transactions.presenter.SpecificTransactionsPresenter;
import org.mifos.mobilewallet.mifospay.transactions.ui.adapter.SpecificTransactionsAdapter;
import org.mifos.mobilewallet.mifospay.utils.Constants;
import org.mifos.mobilewallet.mifospay.utils.RecyclerItemClickListener;

import java.util.ArrayList;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

public class SpecificTransactionsActivity extends BaseActivity implements
        TransactionsContract.SpecificTransactionsView {

    @Inject
    SpecificTransactionsPresenter mPresenter;

    TransactionsContract.SpecificTransactionsPresenter mSpecificTransactionsPresenter;

    @Inject
    SpecificTransactionsAdapter mSpecificTransactionsAdapter;

    @BindView(R.id.rv_transactions)
    RecyclerView mRvTransactions;

    private ArrayList<Transaction> transactions;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_specific_transactions);
        getActivityComponent().inject(this);
        ButterKnife.bind(this);
        mPresenter.attachView(this);
        showBackButton();
        setToolbarTitle(Constants.SPECIFIC_TRANSACTIONS);

        transactions = getIntent().getParcelableArrayListExtra(Constants.SPECIFIC_TRANSACTIONS);

        showProgressDialog(Constants.PLEASE_WAIT);

        setupRecyclerView();

        hideProgressDialog();
    }

    private void setupRecyclerView() {
        mRvTransactions.setLayoutManager(new LinearLayoutManager(this));
        mRvTransactions.setAdapter(mSpecificTransactionsAdapter);
        mRvTransactions.addItemDecoration(new DividerItemDecoration(this,
                DividerItemDecoration.VERTICAL));

        mSpecificTransactionsAdapter.setData(transactions);

        mRvTransactions.addOnItemTouchListener(new RecyclerItemClickListener(this,
                new RecyclerItemClickListener.OnItemClickListener() {
                    @Override
                    public void onItemClick(View childView, int position) {
                        Intent intent = new Intent(SpecificTransactionsActivity.this,
                                ReceiptActivity.class);
                        intent.setData(Uri.parse(
                                Constants.RECEIPT_DOMAIN
                                        + mSpecificTransactionsAdapter.getTransaction(
                                        position).getTransactionId()));
                        startActivity(intent);
                    }

                    @Override
                    public void onItemLongPress(View childView, int position) {

                    }
                }));

    }

    @Override
    public void setPresenter(TransactionsContract.SpecificTransactionsPresenter presenter) {
        mSpecificTransactionsPresenter = presenter;
    }

}

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsActivity.kt
package org.mifospay.history.specific_transactions.ui

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import dagger.hilt.android.AndroidEntryPoint
import com.mifospay.core.model.domain.Transaction
import org.mifospay.R
import org.mifospay.base.BaseActivity
import org.mifospay.history.HistoryContract
import org.mifospay.history.HistoryContract.SpecificTransactionsView
import org.mifospay.history.specific_transactions.presenter.SpecificTransactionsPresenter
import org.mifospay.history.ui.adapter.SpecificTransactionsAdapter
import org.mifospay.receipt.ui.ReceiptActivity
import org.mifospay.common.Constants
import org.mifospay.history.specific_transactions.presenter.SpecificTransactionsViewModel
import org.mifospay.theme.MifosTheme
import org.mifospay.utils.RecyclerItemClickListener
import javax.inject.Inject

@AndroidEntryPoint
class SpecificTransactionsActivity : BaseActivity() {

    private val viewModel: SpecificTransactionsViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel.setArguments(
            transactions = intent.getParcelableArrayListExtra(Constants.TRANSACTIONS)
                ?: arrayListOf(),
            accountNumber = intent.getStringExtra(Constants.ACCOUNT_NUMBER) ?: ""
        )
        setContent {
            MifosTheme {
                SpecificTransactionsScreen(
                    backPress = { onBackPressed() },
                    transactionItemClicked = { transactionId ->
                        navigateToReceiptActivity(transactionId)
                    }
                )
            }
        }
    }

    private fun navigateToReceiptActivity(transactionId: String) {
        val intent = Intent(this@SpecificTransactionsActivity, ReceiptActivity::class.java)
        intent.data = Uri.parse(Constants.RECEIPT_DOMAIN + transactionId)
        startActivity(intent)
    }
}